addons/website/static/src/snippets/s_table_of_content/options.js

```diff
@@ -57,7 +57,17 @@ options.registry.TableOfContent = options.Class.extend({
             $el.attr('id', id);
             $el[0].dataset.anchor = 'true';
         });
-        $nav.find('a:first').addClass('active');
+
+        const tocMainEl = this.$target[0].querySelector('.s_table_of_content_main');
+        if (tocMainEl && tocMainEl.children.length === 0) {
+            this.trigger_up('go_to_parent', {$snippet: this.$target});
+            // destroy public widget and remove the ToC since there are no more
+            // child elements.
+            this.trigger_up('will_remove_snippet', {$target: this.$target});
+            this.$target[0].remove();
+        } else {
+            $nav.find('a:first').addClass('active');
+        }
```

Comment:
As I feared, there's still an error when you delete the last element with the delete key on the keyboard (and [it is no longer linked to summernote this time](https://github.com/odoo/odoo/pull/108374#discussion_r1250857163)). Can you fix this so that when you delete the last element with the keyboard, the block is also deleted (same as with the remove button)? Thank you.